### ISSUE TITLE
Interest in a Debian-version of node-red-docker?

### ISSUE THREAD

**Issue by VincentSC** [node-red/node-red-docker#340]
I ported `node-red-docker` to Docker, and it solved #305 

Is there interest in the port? If yes, I'll share my code.

**Comment by dceejay** [comment]
The main node-red-docker project has instructions here - https://github.com/node-red/node-red-docker/tree/master/docker-custom

**Comment by tomaszduda23** [comment]
Would it be possible to publish official debian version on docker hub? Alpine has a problem with DNS so it would be nice to has debian version also.

**Comment by hardillb** [comment]
Publishing to docker hub comes with doubling the support overhead, so there would need to be a truly compelling reasons to do so.

Do you have a link to the DNS problem you mentioned?

**Comment by tomaszduda23** [comment]
In some cases DNS in alpine just work randomly. When you need to call e.g. some rest API from node-red it is pretty annoying. The easiest way to solve it is debian base image.

https://github.com/alpinelinux/docker-alpine/issues/165
https://gitlab.alpinelinux.org/alpine/tsc/-/issues/43
https://news.ycombinator.com/item?id=28312935
https://news.ycombinator.com/item?id=30642632

**Comment by VincentSC** [comment]
Choosing Debian is not simply for the sake of preference, but for NodeRed to be actually functioning properly.

This is not a preference-discussion like Vim-vs-Emacs. I would therefore stress that it's utterly important to start serving Debian-based dockers on Docker Hub. I actually don't care which more mature Linux is used - as long as it does not have the issues like Alpine has.

As I have written my own docker-build for Debian, I'm totally in the state of WFM ¯\_(ツ)_/¯. But... I would like to see NR to get accepted for low-coding a lot more, and this issue is obstructing that.

**Comment by hardillb** [comment]
There a test debian build (along with all the other 3.1.0-beta.2 builds here 

https://hub.docker.com/r/nodered/node-red-dev/tags

**Comment by VincentSC** [comment]
Tested this setup for a few months:

```
FROM nodered/node-red-dev:v3.1.0-beta.2-debian

USER root

RUN apt update
RUN apt install -y nano iputils-ping
RUN usermod -aG dialout node-red

USER node-red
```

It has been working flawlessly!

**Comment by ingvarso** [comment]
Because of the DNS-resolver issues in Alpine base image, and therefore allmost daily crashes in Node Red, I swiched to building a custom Debian image a month ago. After that, I have not experienced further crashes. In my opinion this is a good reason for also providing a Debian based image at the docker hub.

**Comment by hardillb** [comment]
@ingvarso A debian based build was included in the Node-RED 3.1.0 release yesterday. It is available with the following tags:

- latest-debian
- 3.1.0-debian
- 3.1-debian (will be available once the running respin finishes)

https://hub.docker.com/r/nodered/node-red/tags

It is based on NodeJS 16

**Comment by hardillb** [comment]
Closing as the 3.1 debian build has shipped
